File: src/main/java/no/novari/flyt/archive/gateway/template/validation/AtLeastOneConditionValidator.java
package no.novari.flyt.archive.gateway.template.validation;

import no.novari.flyt.archive.gateway.template.model.ValuePredicate;
import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;

public class AtLeastOneConditionValidator implements HibernateConstraintValidator<AtLeastOneCondition, ValuePredicate> {

    @Override
    public boolean isValid(ValuePredicate value, HibernateConstraintValidatorContext hibernateConstraintValidatorContext) {
        return value.getDefined() != null
                || value.getValue() != null
                || value.getNotValue() != null;
    }

}

File: src/main/kotlin/no/novari/flyt/archive/gateway/template/validation/AtLeastOneConditionValidator.kt
package no.novari.flyt.archive.gateway.template.validation

import no.novari.flyt.archive.gateway.template.model.ValuePredicate
import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext

class AtLeastOneConditionValidator : HibernateConstraintValidator<AtLeastOneCondition, ValuePredicate> {
    override fun isValid(
        value: ValuePredicate?,
        hibernateConstraintValidatorContext: HibernateConstraintValidatorContext,
    ): Boolean = value?.defined != null || value?.value != null || value?.notValue != null
}
